homeassistant/components/homeassistant/scene.py

```diff
@@ -252,6 +259,8 @@ async def create_service(call: ServiceCall) -> None:
         scene_config = SceneConfig(None, call.data[CONF_SCENE_ID], None, entities)
         entity_id = f"{SCENE_DOMAIN}.{scene_config.name}"
         if (old := platform.entities.get(entity_id)) is not None:
+            if TYPE_CHECKING:
```

Comment:
I wonder if `old` could possibly be a `Scene` without being `HomeAssistantScene`, but the current code assumes `from_service` is available.

If it can be a scene, then maybe the check should be moved out of TYPE_CHECKING.

Reply:
I can be any scene entity.

I guess the safest thing to do here is:

```py
        if (old := platform.entities.get(entity_id)) is not None:
            if not isinstance(old, HomeAssistantScene) or not old.from_service:
                _LOGGER.warning("The scene %s already exists", entity_id)
                return
            await platform.async_remove_entity(entity_id)
```